### PR TITLE
CA-236444: WLB consult error when host entry maintenance mode or host

### PR DESCRIPTION
**Pull request by liulinC**
shutdown/reboot and send an empty email (alarm)

Signed-off-by: Lin Liu <lin.liu@citrix.com>

### CODE REVIEW

File: ocaml/xapi/workload_balancing.ml [hunk 1]
```diff
@@ -675,7 +675,7 @@ let get_evacuation_recoms ~__context ~uuid =
       []
     else
       match inner_xml with
-      | (Xml.Element ( _, _, [Xml.Element("CanPlaceAllVMs",_,_)])) ->
+      | (Xml.Element ( _, _, [_;_;Xml.Element("CanPlaceAllVMs",_,_)])) ->
```
Comment (lindig) [inline ocaml/xapi/workload_balancing.ml]: Please expand the commit message. It is not obvious how this change is related to the subject line. You gave an explanation in a comment - this should go into the commit message if it is essential to understand this change.

Comment (thomassa) [inline ocaml/xapi/workload_balancing.ml]: Agreed: I'm sorry but I don't understand what the commit-message means.
Also its first line (subject-line) is awfully long.

Comment (liulinC) [inline ocaml/xapi/workload_balancing.ml]: updated as requested,  could you  please help to review it, thanks.

Comment (thomassa) [inline ocaml/xapi/workload_balancing.ml]: @liulinC commented on the main pull-request to explain that this is to match xml such as
`<HostGetRecommendationsResult i="http://www.w3.org/2001/XMLSchema-instance"> <ErrorMessage nil="true"/> <ResultCode>0</ResultCode> <CanPlaceAllVMs>true</CanPlaceAllVMs> </HostGetRecommendationsResult>`
but this code is fragile: it assumes that `HostGetRecommendationsResult` will have exactly three children, and that `CanPlaceAllVMs` will be the third of them.

The order of those three elements is essentially arbitrary, yes?
So I would prefer the code to work regardless of the order of the elements.
Perhaps something like
```
| (Xml.Element ( _, _, children)) when List.exists
    (function
       | Xml.Element("CanPlaceAllVMs",_,_) -> true
       | _ -> false
    ) children ->
  []
```

**Comment by liulinC** [comment]
update accordingly, please help to review it.

**Pull request merged**